### ISSUE TITLE
Test fail when installing idris2 from Archlinux' AUR

### ISSUE THREAD

**Issue by gyscos** [idris-lang/Idris2#2742]
I am trying to build [`idris2`](https://aur.archlinux.org/packages/idris2) from the AUR (currently version 0.6.0).

The compilation itself looks successful, but when running tests, `base/system_info001` fails. I have a ryzen 5900x with 24 threads, but the test apparently finds 32 processors, while the "expected" value is correctly 24 processors.

```
$ nproc
24
$ nproc --all
32
$ lscpu
Architecture:            x86_64
  CPU op-mode(s):        32-bit, 64-bit
  Address sizes:         48 bits physical, 48 bits virtual
  Byte Order:            Little Endian
CPU(s):                  24
  On-line CPU(s) list:   0-23
Vendor ID:               AuthenticAMD
  Model name:            AMD Ryzen 9 5900X 12-Core Processor
    CPU family:          25
    Model:               33
    Thread(s) per core:  2
    Core(s) per socket:  12
    Socket(s):           1
    Stepping:            0
    Frequency boost:     enabled
    CPU(s) scaling MHz:  90%
    CPU max MHz:         4950.1948
    CPU min MHz:         2200.0000
    BogoMIPS:            7403.53
    Flags:               fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx mmxext fxsr_opt pdpe1gb rdtscp lm constant_tsc rep_good nopl nonsto
                         p_tsc cpuid extd_apicid aperfmperf rapl pni pclmulqdq monitor ssse3 fma cx16 sse4_1 sse4_2 movbe popcnt aes xsave avx f16c rdrand lahf_lm cmp_legacy svm extapic cr8_legacy abm sse
                         4a misalignsse 3dnowprefetch osvw ibs skinit wdt tce topoext perfctr_core perfctr_nb bpext perfctr_llc mwaitx cpb cat_l3 cdp_l3 hw_pstate ssbd mba ibrs ibpb stibp vmmcall fsgsbase
                          bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a rdseed adx smap clflushopt clwb sha_ni xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local clzero irperf x
                         saveerptr rdpru wbnoinvd arat npt lbrv svm_lock nrip_save tsc_scale vmcb_clean flushbyasid decodeassists pausefilter pfthreshold avic v_vmsave_vmload vgif v_spec_ctrl umip pku osp
                         ke vaes vpclmulqdq rdpid overflow_recov succor smca fsrm
Virtualization features: 
  Virtualization:        AMD-V
Caches (sum of all):     
  L1d:                   384 KiB (12 instances)
  L1i:                   384 KiB (12 instances)
  L2:                    6 MiB (12 instances)
  L3:                    64 MiB (2 instances)
NUMA:                    
  NUMA node(s):          1
  NUMA node0 CPU(s):     0-23
Vulnerabilities:         
  Itlb multihit:         Not affected
  L1tf:                  Not affected
  Mds:                   Not affected
  Meltdown:              Not affected
  Mmio stale data:       Not affected
  Retbleed:              Not affected
  Spec store bypass:     Mitigation; Speculative Store Bypass disabled via prctl
  Spectre v1:            Mitigation; usercopy/swapgs barriers and __user pointer sanitization
  Spectre v2:            Mitigation; Retpolines, IBPB conditional, IBRS_FW, STIBP always-on, RSB filling, PBRSB-eIBRS Not affected
  Srbds:                 Not affected
  Tsx async abort:       Not affected
```

```
...
base/system_info001: FAILURE                                     04.255s
Expected:
24 processors

Given:
32 processors
...
```

And indeed, compiling and running this:

```c
#include <unistd.h>
#include <stdio.h>

int main() {
    printf("%i\n", sysconf(_SC_NPROCESSORS_CONF));
    return 0;
}
```
prints 32.

**Comment by CodingCellist** [comment]
Some context which is probably related: https://bbs.archlinux.org/viewtopic.php?id=278874

It seems that `nproc --all` reports the maximum number of processors _supported by the motherboard_, rather than the number of installed cores. This is odd, but would explain the 32 cores, since the 5950x is 16-core/32-thread.

However, I don't know why this causes the test to fail, since we don't use `nproc --all` but just plain `nproc`...

**Comment by CodingCellist** [comment]
@gyscos, could you please check if the fix in the PR above (#2754) works for your system?  : )

**Comment by Blugatroff** [comment]
I have the same problem with the AUR package. 

@CodingCellist I tried to install from source manually using your fix #2754, i followed the instructions in [INSTALL.md](https://github.com/idris-lang/Idris2/blob/main/INSTALL.md) and everything worked perfectly! :D
